wp_api/src/lib.rs

```diff
@@ -212,6 +215,32 @@ pub enum IntegerOrString {
     String(String),
 }
 
+#[uniffi::export(with_foreign)]
+#[async_trait::async_trait]
+pub trait WpAppNotifier: Send + Sync + std::fmt::Debug {
+    async fn unauthorized_request(&self);
+}
+
+pub async fn is_unauthorized_request(
+    status_code: u16,
+    wp_error_code: &WpErrorCode,
+    request_executor: Arc<dyn RequestExecutor>,
+    api_root_url: std::sync::Arc<crate::ParsedUrl>,
+    authentication: crate::WpAuthentication,
+) -> bool {
+    if status_code == 401
+        && (wp_error_code == &WpErrorCode::Unauthorized
+            || wp_error_code == &WpErrorCode::ForbiddenContext)
+    {
+        let r = ApplicationPasswordsRequestBuilder::new(api_root_url, authentication)
+            .retrieve_current_with_edit_context()
+            .into();
+        request_executor.execute(r).await.is_ok()
```

Comment:
Technically, receiving a `RequestExecutorError` does not mean the request is unauthorized. I wonder if we should explicitly check the `WpErrorCode` instead here?